sdbuild/scripts/create_bsp.sh

```diff
@@ -6,6 +6,13 @@ set -e
 board=$1
 template=$2
 
+# If $BSP is a URL, fetch it!
+if [[ "$BSP" == *"://"* ]] ; then
+       BSP_NEW=`basename "$BSP"`
+       BSP_ABS=$PWD/build/$BSP_NEW
```

Comment:
I would refrain from re-defining $BSP_ABS in this way. Better to use known env variables defined in the root Makefile
I would rewrite it as
`BSP_ABS=$BSP_BUILD/../$BSP_NEW`

$BSP_BUILD when launching `create_bsp` will be defined as `$$(BUILD_ROOT_$1)/petalinux_bsp` (see [here](https://github.com/Xilinx/PYNQ/blob/master/sdbuild/Makefile#L98)), so moving a level up will provide the location of the build root for the target board.

Reply:
I've changed BSP_ABS to match what you suggested.

I chose to not create a directory as that opens up many more robustness/reliability issues versus just dropping it in the general BSP directory.